src/blockMonitor/reorgDetector.ts

```diff
@@ -110,6 +116,11 @@ export class ReorgDetector extends StartStopService {
      * @param newHead
      */
     private async conductReorg(newHead: BlockStubChain) {
+        // We ignore further reorgs until this one is complete
+        if (this.conductingReorg) return;
```

Comment:
I dont think this is safe - if two reorgs happen concurrently they will both update the head block but only one of them will fire relevant reorg events

Reply:
You're right; changed it in e31cb01 to stop processing new blocks instead. It still feels unsatisfactory, but I guess further improvemens could be part of #168.

Reply:
Yeah, I agree that this still looks a bit weird. But from what I can see it's safe now. I think we should change the reorg detector logic quite drastically soon, so as long as it's safe lets leave it for now.